### PR TITLE
Update the recommended steps when reporting issues

### PR DESCRIPTION
**Pull request by expez**
Recently quite a few of the bugs reported have been either missing reproduction
steps or been caused by interactions with other packages.

I'm hoping this will increase the percentage of issues which are immediately
actionable without one of us asking some variant of "How can I reproduce this?".

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1078,6 +1078,8 @@ Report issues and suggest features and improvements on the
 questions on the issue tracker - the mailing list and the IRC channel are the
 places for questions.
 
+If you want to file a bug please clone this repo and provide sufficient details to reproduce the issue starting with `make emacs`.  It's likely that your problem is already fixed on `master` or that it's caused by something or other in your `.emacs`.
```
Comment (bbatsov) [inline README.md]: Sounds good, but the name of the task is a bit confusing IMO. It'd be nice if we explained that people are supposed to test cider in isolation from other packages.

**Comment by expez** [comment]
> [...]  the name of the task is a bit confusing IMO.

Yeah, I just went with the convention `evil-mode` uses.  I get of understand the cop-out, finding a good name here is somewhat hard.  Any suggestions? How about `make start-cider`?

**Comment by bbatsov** [comment]
`start-cider` is fine. `run-cider` is also fine.

**Comment by expez** [comment]
Thanks, I went with `run-cider` and updated the readme slightly with the intent.

File: Makefile [hunk 2]
```diff
@@ -47,3 +47,6 @@ elpaclean : clean
 
 %.elc : %.el
 	$(CASK) build
+
+start-cider: elpa
```
Comment (bbatsov) [inline Makefile]: It says `start-cider` here.

File: README.md [hunk 3]
```diff
@@ -1078,6 +1078,9 @@ Report issues and suggest features and improvements on the
 questions on the issue tracker - the mailing list and the IRC channel are the
 places for questions.
 
+If you want to file a bug please clone this repo and provide sufficient details to reproduce the issue starting with `make run-cider`.
+This will start emacs with only the latest version of cider loaded, ensuring that the problem at hand isn't already fixed or caused by interactions with other packages.
```
Comment (bbatsov) [inline README.md]: emacs -> Emacs; cider -> CIDER (for consistency)

File: README.md [hunk 4]
```diff
@@ -1078,6 +1078,9 @@ Report issues and suggest features and improvements on the
 questions on the issue tracker - the mailing list and the IRC channel are the
 places for questions.
 
+If you want to file a bug please clone this repo and provide sufficient details to reproduce the issue starting with `make run-cider`.
```
Comment (bbatsov) [inline README.md]: There should be a `,` after bug. "starting with ..." might be a separate sentence as it reads a bit strange right now. Maybe "Start by running ...".

**Comment by expez** [comment]
updated

**Pull request merged**